tests/integration/test_upgrade_charm.py

```diff
@@ -28,12 +52,77 @@ async def test_deploy_from_edge_and_upgrade_from_local_path(ops_test, prometheus
     """
     async with IPAddressWorkaround(ops_test):
         logger.debug("deploy charm from charmhub")
-        await ops_test.model.deploy(f"ch:{app_name}", application_name=app_name, channel="edge")
-        await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=TIMEOUT)
+        await ops_test.model.deploy(
+            f"ch:{PROMETHEUS_APP_NAME}", application_name=PROMETHEUS_APP_NAME, channel="edge"
+        )
+        await ops_test.model.wait_for_idle(
+            apps=[PROMETHEUS_APP_NAME], status="active", timeout=TIMEOUT
+        )
 
         logger.debug("upgrade deployed charm with local charm %s", prometheus_charm)
-        await ops_test.model.applications[app_name].refresh(
-            path=prometheus_charm, resources=resources
+        await ops_test.model.applications[PROMETHEUS_APP_NAME].refresh(
+            path=prometheus_charm, resources=prometheus_resources
         )
-        await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=TIMEOUT)
-        await check_prometheus_is_ready(ops_test, app_name, 0)
+        await ops_test.model.wait_for_idle(
+            apps=[PROMETHEUS_APP_NAME], status="active", timeout=TIMEOUT
+        )
+        await check_prometheus_is_ready(ops_test, PROMETHEUS_APP_NAME, 0)
+
+
+async def test_metrics_endpoint_provider_can_be_upgraded(ops_test, prometheus_tester_charm):
```

Comment:
```suggestion
async def test_upgrading_rules_provider_also_updates_rule_files(ops_test, prometheus_tester_charm):
```

Reply:
I am fine with changing this title. I just don't want to trigger and test run and wait for re-reviews. I hope it is ok if I just merge as is now and will change the title as part of refactor of other test. Feel free to remind me if I fail to do so.